File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> pos = new HashMap<>();

        for(int i=0;i< nums.length;i++){
            pos.put(nums[i],i);
        }

        for(int i=0;i<nums.length;i++){
            int temp = target-nums[i];
            if(pos.containsKey(temp) && pos.get(temp)!=i){
                return new int[]{pos.get(temp),i};
            }
        }

        return new int[]{-1,-1};
    }

}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsDict = {}
        res = [0,1]
        for i,n in enumerate(nums):
            if target-n not in numsDict:
                numsDict[n]=i
            else:
                return [i, numsDict[target-n]]
        
        return res
        